command/src/channel.rs

```diff
@@ -77,25 +78,27 @@ impl<Tx: Debug + Serialize, Rx: Debug + DeserializeOwned> Channel<Tx, Rx> {
         }
     }
 
-    fn set_nonblocking(&mut self, nonblocking: bool) {
+    fn set_nonblocking(&mut self, nonblocking: bool) -> anyhow::Result<()> {
         unsafe {
             let fd = self.sock.as_raw_fd();
             let stream = net::UnixStream::from_raw_fd(fd);
-            let _ = stream.set_nonblocking(nonblocking).map_err(|e| {
```

Comment:
We should check in which cases this operation can fail and then take a better decision about what we have to do above

Reply:
There is very little chance this operation fails, as we've never seen it happen anywhere.

Or what do you mean, *above*?